pydis_site/apps/api/viewsets/bot/off_topic_channel_name.py

```diff
@@ -106,6 +110,31 @@ def list(self, request: HttpRequest) -> Response:
                     'random_items': ["Must be a positive integer."]
                 })
 
+            if 'mark_used' in request.query_params and request.query_params['mark_used']:
+                queryset = self.get_queryset().order_by('?').exclude(used=True)[:random_count]
+                self.get_queryset().filter(
+                    name__in=(query.name for query in queryset)
+                ).update(used=True)
+
+                # When client request more channel names than non-used names is available, start
+                # new round of names.
+                if len(queryset) < random_count:
+                    # Get how much names still missing and don't fetch duplicate names.
+                    need_more = random_count - len(queryset)
+                    ext = self.get_queryset().order_by('?').exclude(
+                        name__in=(query.name for query in queryset)
+                    )[:need_more]
+
+                    # Set all names `used` field to False except these that we just used.
+                    self.get_queryset().exclude(name__in=(
+                        query.name for query in ext)
+                    ).update(used=False)
+                    # Join original queryset (that had missing names)
+                    # and extension with these missing names.
```

Comment:
```suggestion
                    ).update(used=False)

                    # Join the original queryset (that had missing names)
                    # and the extension with these missing names.
```
Always put a newline before a block comment, if possible.

Also, I'm not sure what you're doing here, can you explain it better?

Reply:
This joins these missing names to the original list of names.